pyca/getfunctions.hh

```diff
@@ -107,7 +108,8 @@ PyObject* _pyca_get_value(capv* pv, const T* dbrv, long count)
         npy_intp dims[1] = {count};
         int typenum = _numpy_array_type(&(dbrv->value));
         PyObject* nparray = PyArray_EMPTY(1, dims, typenum, 0);
-        memcpy(PyArray_DATA(nparray), &(dbrv->value), count*sizeof(dbrv->value));
+        PyArrayObject *arr = (PyArrayObject *)PyArray_FROM_O(nparray); 
```

Comment:
Some notes for the reviewer:  [PyArray_DATA](https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_DATA) expects to take a `PyArrayObject` as its argument.  This wasn't explicitly called out in the[ numpy2.0 migration guide](https://numpy.org/doc/stable/numpy_2_0_migration_guide.html), but was causing compilation errors.